src/components/Button.js

```diff
@@ -41,15 +43,13 @@ module.exports = React.createClass({
 		};
 	},
 	render() {
+		var classes = this.props.sheet.classes;
```

Comment:
`2` I mean that ideally I want to clearly see all props passed to the component via attributes, when I render that component. This is also the way ESLint React plugin checks for undeclared props. You simply cannot inject props into component if you sat that rule. It makes code much cleaner, and is the reason I went for children-as-function approach.
When I open a component and see some props, I go to propTypes.
And then I look at parent component to see what it passes down. I will see nothing. Which means I need to open JSS documentation to see what kind of props it injects. And that adds complexity.